### DISCUSSION TITLE
My model formation

### DISCUSSION THREAD

**Discussion by anujyadav0693** [neperfepx/neper#564] [Q&A] [answered]
respected sir, thanks for providing us this wonderful neper software . 
i want to generate micro pores in my 2d and 3d model. you have already given the answer in discussions#339 but i am not able to make pores in my tessellation model . i do not want to mesh it. please reply.
thank you.

```
neper -T -n from_morpho -morpho "diameq:lognormal(0.1,0.01)+0.5*lognormal(0.01,0.001),1-sphericity:lognormal(0.145,0.03)" -group "diameq>0.03?1:2" -morphooptiini "weight:radeq,coo:packing" -dim 2 -o 2d
```

question1   in this code why 0.5 is multiplied with pores distribution?
question2   what is meaning of diameq>0.03?1:2 ?
question3    what is meaning of this weight:radeq,coo:packing?
question4     how can remove cell  in tessellation -T ?
I use rmcell() but it is not working.
please write a code for this again

**Comment by anujyadav0693** [comment]
sir, please give suggestions.

**Reply by ashokambig** [reply]
please check my answer in#339
May be it helps

**Comment by rquey** [comment, marked as answer]
> question1 in this code why 0.5 is multiplied with pores distribution?

0.5 is a multiplicative factor that applies to the number of cells. `lognormal(0.1,0.01)+0.5*lognormal(0.01,0.001)` means that 2/3 of the grains will  follow the first distribution and 1/3 will follow the second.

> question2 what is meaning of diameq>0.03?1:2 ?

The is the C/C++-style if-then-else: https://neper.info/doc/exprskeys.html?highlight=lazy#ternary-operators.

> question3 what is meaning of this weight:radeq,coo:packing?

This specific value corresponds to running a (rudimental) dense sphere packing algorithm for the initial state. It helps for wide/multimodal size distributions.

> question4 how can remove cell in tessellation -T ? I use rmcell() but it is not working. please write a code for this again

True that it is buggy right now. I'll try to fix it.

**Comment by anujyadav0693** [comment]
Thanku sir for you valuable information.
